### PR TITLE
Add support for HTTPS

### PR DESCRIPTION
**Pull request by funnelfiasco**
We don't break the ability to just use a hostname without a protocol (http is assumed if none is given), and I added a curl-like `-k` option for self-signed certificates, etc.

### CODE REVIEW

File: cycle_submit.py [hunk 1]
```diff
@@ -273,16 +279,20 @@ def main():
 	else:
 		password = options.password
 
+        if options.hostname.startswith('http'):
```
Comment (dougclayton) [inline cycle_submit.py:282]: We should not be overloading https:// into the hostname argument. If we called it --host, it should not have a protocol. But it should also not have a port, so maybe this fits in with the previous poor choice.

File: cycle_submit.py [hunk 2]
```diff
@@ -273,16 +279,20 @@ def main():
...
+            hostname = options.hostname
+        else:
+            hostname = 'http://' + options.hostname
```
Comment (dougclayton) [inline cycle_submit.py:285]: Building up a "url" variable might be better than making a local "hostname" variable that is not a hostname, and also not the same as the value in "options.hostname".

**Comment by funnelfiasco** [comment]
I agree, its overloaded. I can change the internal variables no problem. I'm just concerned about changing the user-facing behavior. We could do something like add `--url-base` as the "real" option, but if the user uses `host` instead, we'll use that but print a deprecation warning. Too complicated for a little tool like this?

**Comment by dougclayton** [comment]
No worries, it's not worth changing.

**Pull request merged**